lib/reek/smells/smell_repository.rb

```diff
@@ -43,25 +42,21 @@ def examine(scope)
       end
 
       def detectors
-        @initialized_detectors ||= begin
-          @detectors = {}
-          smell_types.each do |klass|
-            @detectors[klass] = klass.new(source_via)
-          end
-          @detectors
-        end
+        @initialized_detectors ||= smell_types.map do |klass|
+          { klass => klass.new(source_via) }
+        end.reduce({}, :merge)
       end
 
-      private
+      protected
```

Comment:
>> No, to silence warnings we otherwise throw at anyone who uses the reek API.

Ok, very good point. Then how about switching back to that gem you proposed to introduce before? @mvz WDYT?